### PR TITLE
Add timeout for windows service commands

### LINKED ISSUE #18262: server startWinService & stopWinService commands give incorrect/misleading return codes
**Describe the bug**
Sometimes **server startWinService** command returns rc=1 when service starts successfully (should be rc=0).
Sometimes **server stopWinService** command returns rc=6 when service stops successfully (should be rc=0).

**Steps to Reproduce**
The "start" problem seems to be a timing issue and happens intermittently when starting the service.  Prunsrv returns 0, but then the server.bat script checks if the server is running and finds that it is not.  A second or two later, the server is running.
The "stop" problem is reported by Prunsrv.  I don't know why Prunsrv thinks the service did not stop.  Maybe Prunsrv isn't waiting long enough.   Prunsrv will return rc=6 any time stopping the service fails; that is rc=6 just indicates that the service failed to stop - not why it failed.  The server does seem to successfully stop despite the bad return code.

**Expected behavior**
rc should be 0 when the windows service successfully starts and when it successfully stops.

**Diagnostic information:**
 - OpenLiberty Version: [e.g. 21.0.0.4.  Probably happens on versions prior to this]
 - Java Version: [N/A]
 - server.xml configuration [N/A]

**Additional context**
Customer reported problem occurs when registering windows service with Administrator ID and then switch to another ID with admin privileges.   I recreated without switching users.   Difficult to recreate, but once it fails it seems to fail fairly consistently.    Also note that stopping the service from the Windows services GUI (services.msc) seems to work once after registering the service, but from then on seems to hang.   That is a different problem, but maybe related.

### PR DESCRIPTION
**Pull request by jimblye**
For issue #18262 

Adds configurable time-outs for liberty-windows-service commands:   server startWinService, server stopWinService

### CODE REVIEW

**Review by tbitonti — APPROVED** [review]
Minor nit: I would change:
"waits for the service to start before continuing. Default is 5."
To:
"waits for the service to start before continuing. Default is 5 seconds."

File: dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server.bat [hunk 1]
```diff
@@ -411,7 +426,15 @@ goto:eof
   call:serverExists true
   if %RC% == 2 goto:eof
   "!WLP_INSTALL_DIR!\bin\tools\win\prunsrv.exe" //SS//%SERVER_NAME%
-  set RC=!errorlevel!
+  set PRUNSRV_RC=!errorlevel!
+
+  @rem Wait up to WINDOWS_SERVICE_START_TIMEOUT seconds for server status to be 1, meaning stopped.
+  @rem RC=0 indicates the server is running; ie the stop request failed.
+  @rem      Reset RC to the return code given by prunsrv (expect rc=6 when stop fails).
```
Comment (hlhoots) [inline dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server.bat]: Looks like the text is tabbed over after @rem

File: dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server.bat [hunk 2]
```diff
@@ -411,7 +426,15 @@ goto:eof
...
+  @rem RC=1 is what we are expecting, meaning server running, but change it to RC=0 to indicate success1
```
Comment (hlhoots) [inline dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server.bat]: success1 at end of @rem statement

Comment (jimblye) [reply dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server.bat]: done

File: dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server.bat (hunk 1 above)
Comment (jimblye) [reply dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server.bat]: Actually that was intentional, showing the return codes from serverRunning in the left hand column.   The indent was to show the change to the return code.   However, I didn't indent for the RC=1 return code which also gets changed.
I updated the comment, indenting again under RC=1 for consistency.

**Comment by jimblye** [comment]
#build #libby

**Comment by jimblye** [comment]
#build
#libby

**Comment by jimblye** [comment]
#build #libby

**Review by hlhoots — APPROVED** [review]
LGTM.

**Pull request merged**